lir/config/visualization.py

```diff
@@ -1,30 +1,59 @@
+from collections import ChainMap
 from pathlib import Path
-from typing import Callable
+from typing import Callable, Any
 
 from lir import registry
 from lir.config.base import GenericFunctionConfigParser, YamlParseError, ContextAwareDict
 from lir.registry import ComponentNotFoundError
+from functools import partial
+from collections.abc import Mapping
 
 
-def parse_visualizations(config: ContextAwareDict, output_path: Path) -> list[Callable]:
+def parse_visualizations(config: ContextAwareDict, output_path: Path) -> list[Callable[..., Any]]:
     """Prepare a list of functions to obtain the configured visualizations.
 
     Visualization functions must be available in the registry and accept three arguments:
        - output_dir: Path, the directory where output files can be created
        - lrs: np.ndarray, a list of lrs
        - labels: np.ndarray, a list of labels corresponding to the lrs
+
+    Additionally, any provided extra options are parsed and passed as parameters to the
+    corresponding visualization function. In the example below, the keyword arguments `h1_color`
+    and `h2_color` will be passed to the corresponding function to generate `pav` plots.
+
+    ```yaml
+    visualization:
+      - pav:
+         - h1_color: yellow
+         - h2_color: green
+    ```
     """
-    visualization_functions = []
-    for visualization_type in config:
+    visualization_functions: list[Callable[..., Any]] = []
+
+    # Collect the visualizations declared in the configuration as key value pairs of name/parameters
+    defined_visualizations: dict[str, dict] = {}
```

Comment:
defined visualizations vind ik verwarrend, want wat betekent defined in deze context? Voorstel: simpelweg `visualizations`, of eventueel iets als `requested_visualizations` of `visualization_definitions`